05-onchain.md

```diff
@@ -2,255 +2,294 @@
-contains an output corresponding to the HTLC.
+A node:
+  - if the commitment transaction HTLC output is spent using the payment
+  preimage, the output is considered *irrevocably resolved*:
+    - MUST extract the payment preimage from the transaction input witness.
+  - if the commitment transaction HTLC output has *timed out* and hasn't been
+  *resolved*:
+    - MUST *resolve* the output by spending it using the HTLC-timeout
+    transaction.
+    - once the resolving transaction has reached reasonable depth:
+      - MUST fail the corresponding incoming HTLC (if any).
+      - MUST resolve the output of that HTLC-timeout transaction.
+      - SHOULD resolve the HTLC-timeout transaction by spending it to a
+      convenient address.
+        - Note: if the output is spent (as recommended), the output is
+        *resolved* by the spending transaction, otherwise it is considered
+        *resolved* by the commitment transaction itself.
+      - MUST wait until the `OP_CHECKSEQUENCEVERIFY` delay has passed (as
+      specified by the remote node's `open_channel` `to_self_delay` field)
+      before spending that HTLC-timeout output.
+  - for any committed HTLC that does NOT have an output in this commitment
+  transaction:
+    - once the commitment transaction has reached reasonable depth:
+      - MUST fail the corresponding incoming HTLC (if any).
+    - if no *valid* commitment transaction contains an output corresponding to
+    the HTLC.
+      - MAY fail the corresponding incoming HTLC sooner.
 
 ### Rationale
 
-The payment preimage either serves to prove payment (when the offering node originated the payment) or
-to redeem the corresponding incoming HTLC from another peer (when the offering node is forwarding the payment). Once a node has extracted the payment, it no longer cares about the fate of the HTLC-spending transaction itself.
+The payment preimage either serves to prove payment (when the offering node
+originated the payment) or to redeem the corresponding incoming HTLC from
+another peer (when the offering node is forwarding the payment). Once a node has
+extracted the payment, it no longer cares about the fate of the HTLC-spending
+transaction itself.
 
-In cases where both resolutions are possible (e.g., when a node receives payment success after timeout), either interpretation is
-acceptable; it is the responsibility of the recipient to spend it
-before this occurs.
+In cases where both resolutions are possible (e.g., when a node receives payment
+success after timeout), either interpretation is acceptable; it is the
+responsibility of the recipient to spend it before this occurs.
 
-We need to use our HTLC-timeout transaction to time out the HTLC
-to prevent them fulfilling it and claiming the funds, and before we can back-fail any corresponding incoming HTLC, using
-`update_fail_htlc` (presumably with reason
+We need to use local HTLC-timeout transaction to time out the HTLC to prevent them
```

Comment:
them->"the remote note"? (or maybe the HTLC?)

Reply:
Since I can't easily see the headers here, I might be misreferencing "we" and "them" in this or later comments. The goal is just to get out the last vestiges of these that we've missed and replace them with the correct antecedents.

Reply:
Agreed; I've only edited thoroughly to line 248 (roughly half the document) in this PR, in order to make the review process less burdensome. I will clean up all the we/them stuff in the next PR. Thanks.